core_impl/src/main/java/com/google/instrumentation/trace/SpanImpl.java

```diff
@@ -46,6 +54,15 @@
   // The start time of the span. Set when the span is created iff the RECORD_EVENTS options is
   // set, otherwise 0.
   private final long startNanoTime;
+  // List of the recorded annotations.
+  @GuardedBy("this")
```

Comment:
Not necessary, Users can create a Span object and share it between multiple threads and record events in the same time.